### PR TITLE
Further HCM fixes and update style lint packages to cope with system colours (sort of)

### PR DESCRIPTION
**Pull request by helenb**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/views/4019870)

### Description of Changes Made
Adds link underline, and thicker underline on hover, to navigation items in HCM
Does the same for rich text links in HCM
Makes the 'More' button consistent with the HCM version in the careers site

### How to Test
To test in high contrast mode, ensure 'forced colours: active' has been selected in dev tools under 'rendering'. Then adjust the 'prefers-color-scheme' option to light or dark and test both modes.

Test both the home page and an inner page. To test rich text links, edit a standard page and add a paragraph block and add a link to it.

### Screenshots

<details>
  <summary>Expand to see more</summary>
Please ask if there are any screenshots you'd like to see in particular.
</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [ ] Consider adding unit tests, especially for bug fixes. If you don't, tell us why.
- [x] Tests and linting passes.
- [ ] Consider updating documentation. If you don't, tell us why.
- [x] If relevant, list the environments / browsers in which you tested your changes: Chrome plus dev tools to mimic HCM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,8 +87,8 @@
     "prettier": "^2.5.1",
     "sass": "^1.45.0",
     "sass-loader": "^12.4.0",
-    "stylelint": "^13.0.0",
-    "stylelint-config-torchbox": "^1.0.0",
+    "stylelint": "^13.13.0",
+    "stylelint-config-torchbox": "^1.1.1",
```
Comment (helenb) [inline package.json:91]: This update to `stylelint-config-torchbox` stops the linter complaining about system colours with errors such as `Expected variable, function or keyword for "ButtonText" of "color"`. However, after updating the packages, we do still get errors like `Expected "ActiveText" to be "activetext"       value-keyword-case`. `stylelint` have apparently handled this is a recent release, but there are breaking changes when updating to version 14 which I don't have time to address right now - see https://stylelint.io/migration-guide/to-14/#syntax-option-and-automatic-inferral-of-syntax. So I have simply disabled this rule around system colours for now.

Comment (chris-lawton) [reply package.json:91]: Feel like we need a ticket to update now that the config has been updated - https://github.com/torchbox/stylelint-config-torchbox/releases/tag/v3.0.0

Comment (chris-lawton) [reply package.json:91]: Is it worth disabling `value-keyword-case` in the stylelint config file?

File: tbx/static_src/sass/abstracts/_mixins.scss [hunk 2]
```diff
@@ -193,6 +193,17 @@
     }
 }
 
+@mixin hcm-underline() {
+    text-decoration: underline;
+    text-decoration-color: var(--color--underline);
+    text-underline-offset: 2px;
+    text-decoration-thickness: 2px;
+}
```
Comment (chris-lawton) [inline tbx/static_src/sass/abstracts/_mixins.scss:201]: nice!

File: package.json (hunk 1 above)
Comment (helenb) [reply package.json:91]: I think we'd still want it enabled for the rest of the code.

**Review by chris-lawton — APPROVED** [review]

**Pull request merged**